### PR TITLE
fix logout and signup session

### PR DESCRIPTION
**Pull request by x37v**
I wasn't exactly sure how the testing works, I would have liked to have seen the tests fail for logout as well as signup sessions, but this works in practice and doesn't break the tests.

### CODE REVIEW

File: server/api/logout.js [hunk 1]
```diff
@@ -17,11 +17,13 @@ exports.register = function (server, options, next) {
...
+                return reply({ message: 'Session not found.' }).takeover().code(404);
+            }
+
             var Session = request.server.plugins['hapi-mongo-models'].Session;
-            var credentials = request.auth.credentials || { user: {} };
-            var user = credentials.user || {};
             var query = {
-                username: user.username
+                userId: request.auth.credentials.user._id.toString()
             };
```
Comment (jedireza) [inline server/api/logout.js]: Maybe it's best to follow suite from Aqua here?
https://github.com/jedireza/aqua/blob/master/server/api/logout.js#L25-L41

Comment (x37v) [inline server/api/logout.js]: I don't understand why I'd want to hit the database when I already know that the credentials are bad..

Comment (jedireza) [inline server/api/logout.js]: Yes I see what you mean. How about using `Session.findByIdAndDelete` instead of `Session.deleteMany`?

Comment (x37v) [inline server/api/logout.js]: Ahh, ha.. yeah, missed that part.  Just pushed another change.

**Comment by jedireza** [comment]
Once we get tests passing again, let's squash the commits into one and then we should be good to merge.

**Comment by x37v** [comment]
Sounds good.  Am I correct in thinking that since the behavior has changed, we don't hit the DB, that the 404 is actually expected now instead of 500?  I'm not super clear on how the test setup works, it looks like there is no 'session._id' in the test that is supposed to give 200.

**Comment by jedireza** [comment]
Yes, the tests will need to be updated to support the new behavior. Previously we stubbed the `deleteOne` method of the model to return `0`, which ultimately led the code down a path where the 404 was produced.

https://github.com/jedireza/frame/blob/master/test/server/api/logout.js#L108-L129

With your new short circuit logic, you'll need to stub `request.auth.credentials` in a deterministic way. Although it may give the slightest of optimizations, the original logic may actually be much easier to test.

**Comment by jedireza** [comment]
Closing in favor of #93

**Pull request closed without merging**